### PR TITLE
[TF FE]: Support complex tensors for Rsqrt operations

### PR DESCRIPTION
**Pull request by hub-bla**
### Details:
 - Support complex tensors for `Rsqrt`operations + tests

### Tickets:
 - [None](https://github.com/openvinotoolkit/openvino/issues/23238)

### CODE REVIEW

**Comment by rkazants** [comment]
build_jenkins

**Comment by hub-bla** [comment]
Hey @rkazants, I encountered some issues described [here](https://github.com/openvinotoolkit/openvino/issues/23238#issuecomment-2307224307). Could you give me a hint?

**Comment by hub-bla** [comment]
@rkazants This one really got me sweating but I was able to fix the issues. However, tests cannot compare infinite values even though the outcome is correct.

File: tests/layer_tests/tensorflow_tests/test_tf_Rsqrt.py [hunk 1]
```diff
@@ -56,3 +56,71 @@ def test_rsqrt(self, params, ie_device, precision, ir_version, temp_dir, use_leg
...
+        tf.compat.v1.reset_default_graph()
+
+        # Create the graph and model
+        with tf.compat.v1.Session() as sess:
+            param_real = tf.compat.v1.placeholder(np.float32, shape, 'param_real')
+            param_imag = tf.compat.v1.placeholder(np.float32, shape, 'param_imag')
+
+            complex = tf.raw_ops.Complex(real=param_real, imag=param_imag)
+            
+            result = tf.math.rsqrt(complex, name='Operation')
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_Rsqrt.py:89]:
```suggestion
            result = tf.raw_ops.Rsqrt(x=complex, name='Rsqrt')
```

File: tests/layer_tests/tensorflow_tests/test_tf_Rsqrt.py [hunk 2]
```diff
@@ -56,3 +56,71 @@ def test_rsqrt(self, params, ie_device, precision, ir_version, temp_dir, use_leg
...
+                 dict(shape=[1, 3, 50, 100, 224])
+                ]
+
+    @pytest.mark.parametrize("params", test_data)
+    @pytest.mark.nightly
+    def test_rsqrt(self, params, ie_device, precision, ir_version, temp_dir, use_legacy_frontend):
+        self._test(*self.create_complex_rsqrt_net(**params, ir_version=ir_version,
+                                          use_legacy_frontend=use_legacy_frontend),
+                   ie_device, precision, ir_version, temp_dir=temp_dir,
+                   use_legacy_frontend=use_legacy_frontend)
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_Rsqrt.py:126]: please put all test cases into precommit. We no longer split precommit and nighlty

File: tests/layer_tests/tensorflow_tests/test_tf_Rsqrt.py [hunk 3]
```diff
@@ -56,3 +56,60 @@ def test_rsqrt(self, params, ie_device, precision, ir_version, temp_dir, use_leg
...
+        param_real_shape_1 = inputs_info['param_real:0']
+        param_imag_shape_1 = inputs_info['param_imag:0']
+        
+        inputs_data = {}
+        inputs_data['param_real:0'] =  rng.uniform(-10.0, 10.0, param_real_shape_1).astype(np.float32) 
+        inputs_data['param_imag:0'] = rng.uniform(-10.0, 10.0, param_imag_shape_1).astype(np.float32) 
+        
+        return inputs_data
+
+    def create_complex_rsqrt_net(self, shape, ir_version, use_legacy_frontend):
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_Rsqrt.py:77]:
```suggestion
    def create_complex_rsqrt_net(self, shape):
```

File: tests/layer_tests/tensorflow_tests/test_tf_Rsqrt.py [hunk 4]
```diff
@@ -56,3 +56,60 @@ def test_rsqrt(self, params, ie_device, precision, ir_version, temp_dir, use_leg
...
+                 dict(shape=[1, 3, 50, 100, 224])]
+
+    @pytest.mark.parametrize("params", test_data)
+    @pytest.mark.precommit
+    @pytest.mark.nightly
+    def test_rsqrt(self, params, ie_device, precision, ir_version, temp_dir, use_legacy_frontend):
+        self._test(*self.create_complex_rsqrt_net(**params, ir_version=ir_version,
+                                          use_legacy_frontend=use_legacy_frontend),
+                   ie_device, precision, ir_version, temp_dir=temp_dir,
+                   use_legacy_frontend=use_legacy_frontend)
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_Rsqrt.py:115]:
```suggestion
    @pytest.mark.parametrize('shape', [[1], [1, 3], [2, 3, 22], [1, 3, 10, 22]])
    @pytest.mark.precommit
    @pytest.mark.nightly
    def test_rsqrt(self, shape, ie_device, precision, ir_version, temp_dir, use_legacy_frontend):
        self._test(*self.create_complex_rsqrt_net(shape),
                   ie_device, precision, ir_version, temp_dir=temp_dir,
                   use_legacy_frontend=use_legacy_frontend)

```

**Comment by rkazants** [comment]
build_jenkins

**Comment by rkazants** [comment]
Let us see CI results for now. I will take a look more deeply tomorrow.

Thanks,
Roman

**Comment by rkazants** [comment]
build_jenkins

**Comment by rkazants** [comment]
> @rkazants Previous PR #26475 seems to cause failure on ARM and the same happened PR #26512. Should I make separate PR to mark some test cases as xfail or it can be done here?

Hi @hub-bla, do you know the reason? It is strange to me that this is sporadic since the input data should be the same run-to-run.

**Comment by hub-bla** [comment]
I do not know the reason yet but I viewed failed test cases and the output for the failed test is the same. 

Screenshot 2024-09-11 123748
Screenshot 2024-09-11 123851

However, the test on `Rsqrt PR` failed only on 1 test but in `MO Python Test PR` it failed on 3.

**Comment by hub-bla** [comment]
Here in `Rsqrt` I had a problem with an unstable formula at first (It passed tests for small sizes but was really close on bigger ones). I'll try to increase test case size for `Prod` with bigger shapes to see if the formula itself is stable and then get back to you with the results if that's the issue.

**Comment by hub-bla** [comment]
The only other option I found for this operation is iterative multiplication according to the formula for multiplying two complex numbers but I'm not sure how to do it in the translator.

Formula: `(a + ib) (c + id) = (ac – bd) + i(ad + bc)`

**Comment by hub-bla** [comment]
> > @rkazants Previous PR #26475 seems to cause failure on ARM and the same happened PR #26512. Should I make separate PR to mark some test cases as xfail or it can be done here?
> 
> Hi @hub-bla, do you know the reason? It is strange to me that this is sporadic since the input data should be the same run-to-run.

I once found that [catboost implementation on gpu is non-deterministic](https://github.com/catboost/catboost/issues/1066). Maybe it's also the case here.

**Comment by hub-bla** [comment]
@rkazants Does it require any action from me now?

**Comment by rkazants** [comment]
build_jenkins

**Review by rkazants — APPROVED** [review]

**Pull request merged**